### PR TITLE
cognitive: switching to use the embedded sdk

### PR DESCRIPTION
**Pull request by mbfrahry**
Dependent on #13983

### CODE REVIEW

**Review by katbyte — APPROVED** [review]
LGTM  🍣

File: internal/services/cognitive/cognitive_account_resource.go [hunk 1]
```diff
@@ -687,76 +697,76 @@ func expandCognitiveAccountNetworkAcls(d *pluginsdk.ResourceData) (*cognitiveser
...
 	return &results
 }
 
-func expandCognitiveAccountIdentity(vs []interface{}) (*cognitiveservices.Identity, error) {
+func expandCognitiveAccountIdentity(vs []interface{}) (*identity.SystemUserAssignedIdentityMap, error) {
 	if len(vs) == 0 {
-		return &cognitiveservices.Identity{
-			Type: cognitiveservices.ResourceIdentityTypeNone,
+		return &identity.SystemUserAssignedIdentityMap{
+			Type: "None",
```
Comment (tombuildsstuff) [inline internal/services/cognitive/cognitive_account_resource.go:734]: can we use the constants for this from `identity`?

**Review by tombuildsstuff — COMMENTED** [review]
A few minor comments to fix but this should otherwise be good to go 👍

File: internal/services/cognitive/cognitive_account_resource.go [hunk 2]
```diff
@@ -133,12 +135,12 @@ func resourceCognitiveAccount() *pluginsdk.Resource {
 						"type": {
```
[the remark this answers is not in the document]
Comment (mbfrahry) [reply internal/services/cognitive/cognitive_account_resource.go:135]: I think we'll need to make that. I'm not seeing a common schema in `identity`

File: internal/services/cognitive/cognitive_account_resource.go (hunk 1 above)
Comment (mbfrahry) [reply internal/services/cognitive/cognitive_account_resource.go:734]: We can't as they're local to the `identity` package. Should we make them available outside the `identity` package?

File: internal/services/cognitive/cognitive_account_resource.go [hunk 3]
```diff
@@ -520,135 +522,143 @@ func resourceCognitiveAccountRead(d *pluginsdk.ResourceData, meta interface{}) e
...
+			}
+			if err := d.Set("storage", flattenCognitiveAccountStorage(props.UserOwnedStorage)); err != nil {
+				return fmt.Errorf("setting `storages` for Cognitive Account %q: %+v", id, err)
+			}
+			if props.RestrictOutboundNetworkAccess != nil {
+				d.Set("outbound_network_access_restrited", props.RestrictOutboundNetworkAccess)
+			}
+			if props.DisableLocalAuth != nil {
+				d.Set("local_auth_enabled", !*props.DisableLocalAuth)
+			}
```
[the remark this answers is not in the document]
Comment (mbfrahry) [reply internal/services/cognitive/cognitive_account_resource.go]: Done!

**Review by katbyte — APPROVED** [review]
LGTM 🌈

**Pull request merged**